src/Components/Common/Tables.js

```diff
@@ -39,4 +51,133 @@ const passFilters = (rule, filters) =>
     }
   });
 
-export { passFilters };
+const passFiltersCluster = (cluster, filters) =>
+  Object.entries(filters).every(([filterKey, filterValue]) => {
+    switch (filterKey) {
+      case 'text':
+        return (cluster.cluster_name || cluster.cluster_id)
+          .toLowerCase()
+          .includes(filterValue.toLowerCase());
+      case CLUSTER_FILTER_CATEGORIES.hits.urlParam:
+        return (
+          // clusters with at least one rule hit
+          (filterValue.length === 0 && parseInt(cluster.total_hit_count) > 0) ||
+          // all clusters
+          filterValue.includes('all') ||
+          // clusters with at least one rule hit for any of the active risk filters
+          filterValue.some((v) => cluster.hits_by_total_risk[v - 1] > 0)
+        );
+      default:
+        return true;
+    }
+  });
+
+const mapClustersToRows = (clusters) =>
+  clusters.map((c, i) => ({
+    cluster: c,
+    cells: [
+      <span key={i}>
+        <Link to={`clusters/${c.cluster_id}`}>
+          {c.cluster_name || c.cluster_id}
+        </Link>
+      </span>,
+      c.total_hit_count,
+      c.hits_by_total_risk?.[3] || 0,
+      c.hits_by_total_risk?.[2] || 0,
+      c.hits_by_total_risk?.[1] || 0,
+      c.hits_by_total_risk?.[0] || 0,
+      <span key={i}>
+        {c.last_checked_at ? (
+          <DateFormat
+            extraTitle={`${intl.formatMessage(messages.lastSeen)}: `}
+            date={c.last_checked_at}
+            variant="relative"
+          />
+        ) : (
+          <Tooltip
+            key={i}
+            content={
+              <span>
+                {intl.formatMessage(messages.lastSeen) + ': '}
+                {intl.formatMessage(messages.nA)}
+              </span>
+            }
+          >
+            <span>{intl.formatMessage(messages.nA)}</span>
+          </Tooltip>
+        )}
+      </span>,
+    ],
+  }));
+
+const capitalize = (string) => string[0].toUpperCase() + string.substring(1);
+
+const pruneFilters = (localFilters, filterCategories) => {
+  const prunedFilters = Object.entries(localFilters);
+  return prunedFilters.length > 0
+    ? prunedFilters.reduce((arr, item) => {
```

Comment:
No need to check for `length > 0` if the array is empty it will not reduce it. However if the `localFilters` can be undefined you should check for this. The first and second item is used numberous times, it would be better to use `key` and `item` instead.
```suggestion
  return Object.entries(localFilters ?? {}).reduce((arr, [key, item]) => {
```